gcovr/merging.py

```diff
@@ -468,7 +521,33 @@ def merge_condition(
         return left
 
     if left.count != right.count:
-        raise AssertionError("The number of conditions must be equal.")
+        if options.cond_opts.merge_condition_fold:
+            LOGGER.warning(
+                f"Condition counts are not equal, got {left.count} and {right.count}. "
+                f"Reducing to {min(left.count, right.count)}."
+            )
+            if left.count > right.count:
+                left.not_covered_true = left.not_covered_true[
+                    : len(right.not_covered_true)
+                ]
+                left.not_covered_false = left.not_covered_false[
+                    : len(right.not_covered_false)
+                ]
+                left.count = right.count
+            else:
+                right.not_covered_true = right.not_covered_true[
+                    : len(left.not_covered_true)
+                ]
+                right.not_covered_false = right.not_covered_false[
+                    : len(left.not_covered_false)
+                ]
+                right.count = left.count
+        else:
+            raise AssertionError(
+                f"The number of conditions must be equal, got {left.count} and {right.count}.\n"
+                "\tYou can run gcovr with --merge-mode-conditions=COND_MERGE_MODE.\n"
+                "\tThe available values for COMD_MERGE_MODE are described in the documentation."
```

Comment:
Now it's tricky because I added the line information at the end of the exception text. This would look like:
```
The number of conditions must be equal, got x and y.
\tYou can run gcovr with --merge-mode-conditions=COND_MERGE_MODE.
\tThe available values for COMD_MERGE_MODE are described in the documentation. while merging line z.
```
The only thing would be to split the line when extending the message and join it again.

Reply:
Yes I see what you are saying. Will have to fiddle with things slightly.